### PR TITLE
Implementation for Fiat On Ramp

### PR DESCRIPTION
**Pull request by babadrape**
- [ ] Add `Add Funds` button

### CODE REVIEW

File: src/hooks/useSelectNetwork.ts [hunk 1]
```diff
@@ -2,47 +2,35 @@ import { useCallback } from 'react'
...
-            console.warn(error)
-          }
+        } catch (error) {
+          console.warn(error)
         }
       } else {
         dispatch(fetchUserNetwork(chainId, account, userChainId))
       }
     },
-    [chainId, account, provider, dispatch],
```
Comment (Harambe-Nakamoto) [inline src/hooks/useSelectNetwork.ts:45]: This function should be the same as current prod with the additional torus wallet check at the top

File: package.json [hunk 2]
```diff
@@ -29,17 +29,18 @@
     ]
   },
   "dependencies": {
-    "@ape.swap/uikit": "1.0.6",
-    "@apeswapfinance/sdk": "^2.0.0-alpha.14",
+    "@ape.swap/uikit": "^1.0.9-alpha.0",
```
Comment (Harambe-Nakamoto) [inline package.json:32]: Publish uikit as 1.0.9

File: src/hooks/useSelectNetwork.ts (hunk 1 above)
Comment (babadrape) [reply src/hooks/useSelectNetwork.ts:45]: Are you saying this function should have the exact content on production with the addition of the toast check?

File: src/hooks/useSelectNetwork.ts [hunk 3]
```diff
@@ -2,16 +2,26 @@ import { useCallback } from 'react'
...
+  const { chainId, account, library, connector } = useWeb3React()
   const provider = library?.provider
   const dispatch = useDispatch()
+  const { toastError } = useToast()
+  const { t } = useTranslation()
+
   const switchNetwork = useCallback(
     async (userChainId: number) => {
+      if (connector instanceof TorusConnector) {
+        toastError(t('Torus wallet is only available on BSC chain'))
```
Comment (Harambe-Nakamoto) [inline src/hooks/useSelectNetwork.ts:22]: Can we change this to BNB chain

File: package.json [hunk 4]
```diff
@@ -29,17 +29,18 @@
     ]
   },
   "dependencies": {
-    "@ape.swap/uikit": "1.0.6",
+    "@ape.swap/uikit": "^1.0.9",
+    "@apeswapfinance/uikit": "0.11.3-0.3",
```
Comment (Harambe-Nakamoto) [inline package.json:33]: Should we keep this at 0.11.3 or does the fiat onramp rely on 0.11.3-0.3?

**Review by Harambe-Nakamoto — APPROVED** [review]
Nice work!

**Pull request merged**